gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/PluginController.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.controllers;
+
+import com.netflix.spinnaker.gate.services.TaskService;
+import com.netflix.spinnaker.gate.services.internal.Front50Service;
+import com.netflix.spinnaker.kork.web.exceptions.InvalidRequestException;
+import com.netflix.spinnaker.security.AuthenticatedRequest;
+import io.swagger.annotations.ApiOperation;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/pluginArtifacts")
```

Comment:
Perhaps just me but I'm slightly confused at `artifacts` vs `metadata`. 

This API is rooted at `/pluginArtifacts` but all the endpoints speak to dealing with metadata. Is there a difference? 

Can we do better than `List<Map>` everywhere? 

Thoughts on using an envelope object for the `List` response? We've talked about a desire to follow the Google API standards and this is perhaps an opportunity.

Reply:
> This API is rooted at `/pluginArtifacts` but all the endpoints speak to dealing with metadata. Is there a difference?

I was going mostly with how Front50 was exposing those API's . IMO, these are not plugin artifacts that we are trying to persist but the metadata of a plugin. @robzienert : thoughts ?

Reply:
> Thoughts on using an envelope object for the `List` response? We've talked about a desire to follow the Google API standards and this is perhaps an opportunity.

We can do wrapper response objects and to make it fully typed we need to copy the model objects in to this app. Not sure how we feel about replicating model objects into this. ?

Reply:
We can change the model names if we feel like that makes more sense. I was struggling to figure out the right names for this. I followed with calling it an artifact, after kork artifacts, which is just metadata for their binary artifacts.

Reply:
Changed the Controller endpoint to '/pluginInfo'